processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -38,7 +38,11 @@ public int compare(String str1, String str2)
     {
       int[] pos = {0, 0};
 
-      if (str1.length() == 0) {
+      if (str1 == null) {
```

Comment:
this part is purely about the string lengths, `Ordering.nullsFirst` does a lot more than just that, so I don't think we should use it here.